src/controllers/user.js

```diff
@@ -1,20 +1,57 @@
+const UserModel = require('../models/user');
+
 module.exports = {
-  getAllUsers: (req, res) => {
-    res.send('All users');
+  getAllUsers: async (req, res) => {
+    try {
+      const users = await UserModel.find();
+      res.json(users);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
   },
-  getSingleUser: (req, res) => {
-    res.send('Single user');
+  getSingleUser: async (req, res) => {
+    const { id } = req.params;
+    try {
+      const user = await UserModel.findById(id);
+      if (!user) throw new Error("The user with the specified ID wasn't found");
+      res.json(user);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
   },
-  updateUser: (req, res) => {
-    res.send('User with the specified ID has been updated');
+  updateUser: async (req, res) => {
+    const { id } = req.params;
+    try {
+      const updatedUser = await UserModel.findByIdAndUpdate(
+        id,
+        {
+          $set: req.body,
+        },
+        {
+          new: true,
+        }
+      );
+
+      if (!updatedUser)
+        throw new Error("The user with the specified ID wasn't found");
+
+      res.json(updatedUser);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
```

Comment:
You should change this to `message: err.message ?? err`
Because when you're throwing custom errors you're only throwing a string, there is no message property. But some mongoose errors might have the message proeprty.